src/AutoRest.CSharp/Common/Output/Models/RestClientBuilder.cs

```diff
@@ -108,6 +109,16 @@ public RestClientMethod BuildRequestMethod(Operation operation, ServiceRequest s
             );
         }
 
+        public static PathSegment[] BuildRequestPathSegments(Operation operation, ServiceRequest serviceRequest, HttpRequest httpRequest, RestClientBuilder restClientBuilder)
```

Comment:
Method signature where type instance is passed into a static method of that same type is quite unnatural. This should be a normal instance method.